airbyte-cdk/python/airbyte_cdk/sources/streams/http/rate_limiting.py

```diff
@@ -23,21 +23,31 @@
 logger = logging.getLogger("airbyte")
 
 
-def default_backoff_handler(max_tries: Optional[int], factor: float, **kwargs):
-    def log_retry_attempt(details):
+SendRequestCallableType = Callable[[PreparedRequest, Mapping[str, Any]], Response]
+
+
+def default_backoff_handler(
+    max_tries: Optional[int], factor: float, **kwargs: Any
+) -> Callable[[SendRequestCallableType], SendRequestCallableType]:
+    def log_retry_attempt(details: Mapping[str, Any]) -> None:
         _, exc, _ = sys.exc_info()
-        if exc.response:
+        if isinstance(exc, RequestException) and exc.response:
             logger.info(f"Status code: {exc.response.status_code}, Response Content: {exc.response.content}")
         logger.info(
             f"Caught retryable error '{str(exc)}' after {details['tries']} tries. Waiting {details['wait']} seconds then retrying..."
         )
 
-    def should_give_up(exc):
+    def should_give_up(exc: Exception) -> bool:
```

Comment:
I think in practice this is always a RequestException because we only use the backoff to retry HttpRequests, but the [give_up param passed to the backoff library](https://github.com/airbytehq/airbyte/pull/28550/files#diff-21fbde7917b82829112c58ad78af6ca62c2d2ba7229b0318a930176d68be6081R57)  should be a `Callable[[Exception], bool]`